server.js

```diff
@@ -23,47 +23,43 @@ app.use(cookieParser());
 
 // get requests
 app.get("/", async (req, res) => {
-    const { cookies } = req
     try {
-        const currentSession = await Session.findById(cookies.session_id)
-        if (currentSession === null) {
-            res.render('index', currentSession);
+        if (!req.cookies.session) {
+            res.render('index');
         } else {
-            res.render('index', currentSession);
+            const currentUser = await User.findById(req.cookies.session.user);
+            if (currentUser) {
+                res.redirect(`profile/${req.cookies.session.user}`);
```

Comment:
If a user can only access their own profile, why even add the user ID to the url? You know which user it is by their session